torchvision/prototype/transforms/functional/_color.py

```diff
@@ -38,7 +71,19 @@ def adjust_saturation(inpt: features.InputTypeJIT, saturation_factor: float) ->
         return adjust_saturation_image_pil(inpt, saturation_factor=saturation_factor)
 
 
-adjust_contrast_image_tensor = _FT.adjust_contrast
+def adjust_contrast_image_tensor(image: torch.Tensor, contrast_factor: float) -> torch.Tensor:
+    if contrast_factor < 0:
+        raise ValueError(f"contrast_factor ({contrast_factor}) is not non-negative.")
+
+    c = get_num_channels_image_tensor(image)
+    if c not in [1, 3]:
+        raise TypeError(f"Input image tensor permitted channel values are {[1, 3]}, but found {c}")
+    dtype = image.dtype if torch.is_floating_point(image) else torch.float32
+    grayscale_image = _rgb_to_gray(image) if c == 3 else image
+    mean = torch.mean(grayscale_image.to(dtype), dim=(-3, -2, -1), keepdim=True)
```

Comment:
@pmeier actually, doing so the output of `mean` is not the same for uint8 image.
As we originally applied uint8 cast in the end of `_rgb_to_gray` we get rid of all floating point values. This is not the case if image is casted to float before `_rgb_to_gray`. 
So, finally consistency tests report for example:
```
Mismatched elements: 256 / 2772 (9.2%)                                                                                                                                      
Greatest absolute difference: 3 at index (1, 2, 4, 21) (up to 1e-05 allowed)   
Greatest relative difference: 0.25 at index (2, 0, 6, 21) (up to 1e-05 allowed)
```
and this is a real failure, IMO.

Reply:
I agree that the behavior changes, but IMO repeatedly converting to `uint8` in the computation and thus eliminating intermediate values sounds more like a missed opportunity in the original kernel than a bug now. Thus, I would consider this more like a "bug fix" rather than a BC breaking change. On the other hand, that is not a strong opinion. Not going to block over this.